x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_edit.ts

```diff
@@ -49,7 +52,16 @@ export const applyBulkActionEditToRule = (
         "Index patterns can't be added. Machine learning rule doesn't have index patterns property"
       );
 
+      if (
+        isEqlParams(rule.params) ||
+        isQueryParams(rule.params) ||
+        isThresholdParams(rule.params) ||
+        isThreatParams(rule.params)
+      ) {
+        rule.params.dataViewId = undefined;
```

Comment:
This will now wipe data views each time, no longer honoring the API's overwrite dataviews param I think. Is that intentional?

Reply:
Ah yeah this is a bug from when I tried to fix something else in your PR where the index patterns wouldn't be overwritten unless that checkbox was clicked. I have a fix for that here along with passing tests:

a8a8d980449464bbfc506c88160a5e00e9bf4946